src/components/Vqb.vue

```diff
@@ -1,32 +1,108 @@
 <template>
-  <div class="weaverbird">
-    <PipelineSelector class="weaverbird__pipeline-selector" />
-    <ResizablePanels>
-      <QueryBuilder slot="left-panel" />
-      <DataViewer slot="right-panel" />
-    </ResizablePanels>
-  </div>
+  <ResizablePanels>
+    <QueryBuilder slot="left-panel" />
+    <DataViewer slot="right-panel" />
+  </ResizablePanels>
 </template>
 
 <script lang="ts">
 import Vue from 'vue';
-import { Component } from 'vue-property-decorator';
+import { Component, Prop, Watch } from 'vue-property-decorator';
 
 import DataViewer from '@/components/DataViewer.vue';
-import PipelineSelector from '@/components/PipelineSelector.vue';
 import QueryBuilder from '@/components/QueryBuilder.vue';
 import ResizablePanels from '@/components/ResizablePanels.vue';
+import { Pipeline } from '@/lib/steps.ts';
+import { InterpolateFunction } from '@/lib/templating';
+import { registerModule, unregisterModule, VQBModule, VQBnamespace } from '@/store';
+import { VQBState } from '@/store/state';
 
 @Component({
   name: 'vqb',
   components: {
     DataViewer,
-    PipelineSelector,
     QueryBuilder,
     ResizablePanels,
   },
 })
-export default class Vqb extends Vue {}
+export default class Vqb extends Vue {
+  @Prop({
+    type: Object,
+    required: true,
+  })
+  pipelines!: { [name: string]: Pipeline };
```

Comment:
Having a property here and in the store is kinda weird: which one holds the truth? The current prop value? The store?
The VQB component has authority on its store, but not on its props (this belongs to is parent).
Keeping the two in sync would mean watching the store state, emitting events from the VQB to ask the parent component to update the prop (upward), and when the prop changes, change the store value accordingly (downward). To avoid any loop, we would need to be careful about sharing the object references and break the loop when they are the same... This seems a bit complicated, doesn't it? ;)

So let's represent the store elements like this component's data instead of a prop.
In that case, the prop here is not "pipelines", but "initialPipelines". When the component instantiate, it loads into its store this pipelines. They are not to be modified after by the parent.
When working in the VQB, the pipelines in the store gets regularly updated.
We can still emit them to a parent component, but the prop initialPipelines will never be read again to avoid any loop.

(Note: it would not be impossible to achieve this kind of sync IMO. A simple example can be found in https://github.com/ToucanToco/tucana/blob/master/src/scripts/studio/utils/tc-code-editor/monaco/TcMonacoEditor.vue#L121 but it's already not trivial, and it's only for a very simple case with only one prop "value". This is what makes me think to other options).